### PR TITLE
Option to turn off completion brackets

### LINKED ISSUE #1508: sys.stdin tab completes to sys.stdin[
`sys.stdin` tab completes to `sys.stdin[`. I don't see why you would ever want to automatically complete a square bracket, but for sys.stdin it especially doesn't make sense, since it isn't indexable.

### PR DESCRIPTION
**Pull request by gforsyth**
Add `$COMPLETIONS_BRACKETS` (default True) to determine if the python
completer should add opening parens and square brackets to its
completions.

Whether or not `$COMPLETION_BRACKETS` is True, `sys.stdin`, `sys.stdout`
and `sys.stderr` do not have brackets in their completions.

Resolves #1508

### CODE REVIEW

File: xonsh/environ.py [hunk 1]
```diff
@@ -244,6 +245,7 @@ def DEFAULT_VALUES():
         'CDPATH': (),
         'COLOR_INPUT': True,
         'COLOR_RESULTS': True,
+        'COMPLETIONS_BRACKETS': True,
```
Comment (scopatz) [inline xonsh/environ.py]: Can you also add a VarDocs entry for this, please too?

**Comment by scopatz** [comment]
Thanks for this @gforsyth!  Just the one minor issue.  Probably long overdue itself :)

**Comment by gforsyth** [comment]
Hey @scopatz -- ready for another look!

**Comment by scopatz** [comment]
Looks great! Just waiting for CI

**Comment by laerus** [comment]
`/home/travis/build/xonsh/xonsh/xonsh/completers/python.py:99:25: E127 continuation line over-indented for visual indent` (can we turn this off for everything?)

also MacOS builds seem to take very long.

File: xonsh/completers/python.py [hunk 2]
```diff
@@ -95,10 +95,14 @@ def attr_complete(prefix, ctx, filter_func):
...
         a = getattr(val, opt)
-        if callable(a):
-            rpl = opt + '('
-        elif isinstance(a, abc.Iterable):
-            rpl = opt + '['
+        if (builtins.__xonsh_env__['COMPLETIONS_BRACKETS'] and
+                        opt not in ['stdin', 'stdout', 'stderr']):
+            if callable(a):
+                rpl = opt + '('
+            elif isinstance(a, abc.Iterable):
```
Comment (scopatz) [inline xonsh/completers/python.py]: Actually, now that I think about it, this should `isinstance(a, (abc.Sequence, abc.Mapping))` and the special case file handles above should be removed.  We only want to add an `[` when there is a `__getitem__`.  As a counter example, this implementation will still add an `[` to sets, file handles, and other iterable but non-indexable  objects.

**Comment by scopatz** [comment]
> also MacOS builds seem to take very long.

Yeah, Mac has been slow all day.  I noticed it on conda-forge ~18 hrs ago

**Comment by scopatz** [comment]
Thanks a million @gforsyth!

**Pull request merged**